src/ipfs.rs

```diff
@@ -17,14 +17,14 @@ pub async fn fetch_specification_with_retry(
 ) -> anyhow::Result<Specification> {
     let fetch = || async {
         let response = match ipfs_http_client
-            .request(Method::GET, format!("/api/v0/get?arg={cid}"))?
+            .request(Method::POST, format!("/api/v0/cat?arg={cid}"))?
```

Comment:
Why the API change here?

Reply:
Well the method change is because we're actually interacting with a JSON RPC API, so POST must always be used. As for the actual command change, `get` apparently returns other more stuff than simply the content of the file, so the JSON deserialization wasn't working good